File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    /*
        map.containsKey(Key)
        map.put(key, value)
        map.get(key)
    */
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> seen = new HashMap<Integer, Integer>();
        int[] result = new int[2];
        for (int i = 0; i < nums.length; i++) {
            int remains = target - nums[i];
            if (seen.containsKey(remains)) {
                int num = seen.get(remains);
                result[0] = num;
                result[1] = i;
                break;
            }
            seen.put(nums[i], i);
        }
        return result;
    }
}

File: Data Structures & Algorithms/is-anagram/submission-1.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        if (s.Length != t.Length) {
            return false;
        }
        var counter = new Dictionary<char, int>();
        for (int i = 0; i < s.Length; i++) {
            char c = s[i];
            if (counter.ContainsKey(c)) {
                counter[c] = counter[c] + 1;
            } else {
                counter[c] = 1;
            }
        }

        for (int j = 0; j < t.Length; j++) {
            char c = t[j];
            if (counter.ContainsKey(c)) {
                if (counter[c] - 1 < 0) {
                    return false;
                }
                counter[c] = counter[c] - 1;
            } else {
                return false;
            }
        }
        return true;
    }
}
